packages/filecoin-actor-utils/src/utils/params.ts

```diff
@@ -32,3 +34,30 @@ export const describeMessageParams = (
   // Return the described params
   return dataType
 }
+
+// TODO: (nice to have) enable typechain artifact as a typescript generic here so the described params are typed?
+export const describeFEVMMessageParams = (
+  messageParams: string,
+  abi: ABI
+): DataType => {
+  const iface = new ethers.utils.Interface(abi)
+  const paramsHex = cborToHex(messageParams)
+  const parsed = iface.parseTransaction({ data: paramsHex })
+
+  // TODO: how to handle solidity structs as inputs? I believe structs are encoded as `tuple` type https://docs.soliditylang.org/en/v0.5.3/abi-spec.html#mapping-solidity-to-abi-types
+  // TODO: unstandardized capitalization from built-in actors
+  const children = parsed.functionFragment.inputs.reduce<
```

Comment:
This function is incomplete and only works with a depth of 1, where all the DataTypes have a `Value1 (i.e. no solidity array or tuple (struct) types). We should improve it to appropriately handle all the types we need